src/internal/document_api/mod.rs

```diff
@@ -1054,4 +1343,105 @@ mod tests {
+        let plaintext = recrypt_api.gen_plaintext();
+        let encrypted_value = recrypt_api
+            .encrypt(&plaintext, &pubk, &signing_keys)
+            .unwrap();
+        let user_str = "userid".to_string();
+
+        let edek = EncryptedDek {
+            encrypted_dek_data: encrypted_value.clone(),
+            grant_to: WithKey {
+                public_key: pubk.into(),
+                id: UserId::unsafe_from_string(user_str.clone()).borrow().into(),
+            },
+        };
+
+        let proto_edek: EncryptedDekP = edek.borrow().try_into().unwrap();
+
+        assert_eq!(
+            &user_str,
+            &proto_edek.get_userOrGroup().get_userId().to_string()
+        );
+        let (x, y) = pubk.bytes_x_y();
+        assert_eq!(
+            (x.to_vec(), y.to_vec()),
+            (
+                proto_edek
+                    .get_userOrGroup()
+                    .get_masterPublicKey()
+                    .get_x()
+                    .to_vec(),
+                proto_edek
+                    .get_userOrGroup()
+                    .get_masterPublicKey()
+                    .get_y()
+                    .to_vec()
+            )
+        );
+
+        if let recrypt::api::EncryptedValue::EncryptedOnceValue {
+            ephemeral_public_key,
+            encrypted_message,
+            auth_hash,
+            public_signing_key,
+            signature,
+        } = encrypted_value
+        {
+            assert_eq!(
+                (
+                    ephemeral_public_key.bytes_x_y().0.to_vec(),
+                    ephemeral_public_key.bytes_x_y().1.to_vec()
+                ),
+                (
+                    proto_edek
+                        .get_encryptedDekData()
+                        .get_ephemeralPublicKey()
+                        .get_x()
+                        .to_vec(),
+                    proto_edek
+                        .get_encryptedDekData()
+                        .get_ephemeralPublicKey()
+                        .get_y()
+                        .to_vec()
+                )
+            );
+
+            assert_eq!(
+                encrypted_message.bytes().to_vec(),
+                proto_edek
+                    .get_encryptedDekData()
+                    .get_encryptedMessage()
+                    .to_vec()
+            );
+
+            assert_eq!(
+                auth_hash.bytes().to_vec(),
+                proto_edek.get_encryptedDekData().get_authHash().to_vec()
+            );
+
+            assert_eq!(
+                public_signing_key.to_bytes(),
+                proto_edek
+                    .get_encryptedDekData()
+                    .get_publicSigningKey()
+                    .to_vec()
+            );
+
+            assert_eq!(
+                signature.bytes().to_vec(),
+                proto_edek.get_encryptedDekData().get_signature().to_vec()
+            );
+        } else {
+            panic!("Should be EncryptedOnceValue");
```

Comment:
Shouldn't this just be a test failure not a panic?

Reply:
`assert!()` is just a `panic!` with a boolean check, so I think it's okay. Maybe there is just a fail macro? I'll check.